include/revng/Support/IntegerSerialization.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/Support/Endian.h"
+
+#include "revng/ADT/CompilationTime.h"
+#include "revng/ADT/Concepts.h"
+
+namespace detail {
+
+inline constexpr auto IntEndianness = llvm::support::little;
```

Comment:
1. This is an *opinonated* integer serialization library. For instance, it serializes things in little endian. Add documentation about the format we're making up, assume someone has to implement a compatible parser.
   Don't assume this is obvious.
2. Expand all `Int` to `Integer`